File: 2019-06/136-single-number/elixir/136.exs
#! /usr/bin/env elixir
defmodule Solution do
  use Bitwise

  def single_number(nums) when is_list(nums) do
    nums
    |> Enum.reduce(fn x, acc -> x ^^^ acc end)
  end
end

IO.puts("Expect to see: 4")

[4, 1, 2, 1, 2]
|> Solution.single_number()
|> IO.puts()

File: 2020-02/136.py
#!/usr/bin/env python

from typing import List
from functools import reduce
from operator import xor

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        return reduce(xor, nums)

s = Solution()
got = s.singleNumber([2, 2, 1])
want = 1
print(f"got {got} want {want} pass {got == want}")
got = s.singleNumber([4, 1, 2, 1, 2])
want = 4
print(f"got {got} want {want} pass {got == want}")
